src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -28,62 +29,131 @@ internal abstract partial class AbstractInitializeParameterCodeRefactoringProvid
         protected abstract SyntaxNode GetBody(SyntaxNode functionDeclaration);
         protected abstract SyntaxNode GetTypeBlock(SyntaxNode node);
 
+        protected virtual ImmutableArray<SyntaxNode> GetParameters(SyntaxNode parameterNode)
+        {
+            return default;
+        }
         protected abstract void InsertStatement(
             SyntaxEditor editor, SyntaxNode functionDeclaration, IMethodSymbol method,
             SyntaxNode statementToAddAfterOpt, TStatementSyntax statement);
 
-        protected abstract Task<ImmutableArray<CodeAction>> GetRefactoringsAsync(
+        protected abstract Task<ImmutableArray<CodeAction>> GetRefactoringsForAllParametersAsync(
             Document document, IParameterSymbol parameter, SyntaxNode functionDeclaration, IMethodSymbol method,
             IBlockOperation blockStatementOpt, CancellationToken cancellationToken);
 
         public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
-            var (document, textSpan, cancellationToken) = context;
+            var position = context.Span.Start;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
             var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(position);
+            var firstparameterNode = GetParameterNode(token, position);
 
-            var parameterNode = await context.TryGetRelevantNodeAsync<TParameterSyntax>().ConfigureAwait(false);
-            if (parameterNode == null)
+            var parameterNodes = GetParameters(firstparameterNode);
+            if (parameterNodes.IsDefault)
             {
                 return;
             }
 
-            var functionDeclaration = parameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
-            if (functionDeclaration == null)
+            // List with parameterNodes that pass all checks
+            var listOfPotentiallyValidParametersNodes = new List<SyntaxNode>();
+            var counter = 0;
+
+            foreach (var parameterNode in parameterNodes)
             {
-                return;
-            }
+                ++counter;
+                if (parameterNode == null)
```

Comment:
when does this happen?

Reply:
I'm not sure. You think it could never happen?

Reply:
if this was grabbed off of some real member, i can't see hwo this could happen :)